src/fabric.c

```diff
@@ -99,13 +125,42 @@ static int ofi_is_hook_prov(const struct fi_provider *provider)
 
 int ofi_apply_filter(struct fi_filter *filter, const char *name)
 {
-	if (filter->names) {
-		if (ofi_find_name(filter->names, name) >= 0)
-			return filter->negated ? 1 : 0;
+	if (!filter->names)
+		return 0;
 
-		return filter->negated ? 0 : 1;
-	}
-	return 0;
+	if (ofi_find_name(filter->names, name) >= 0)
+		return filter->negated ? 1 : 0;
+
+	return filter->negated ? 0 : 1;
+}
+
+int ofi_apply_prov_init_filter(struct fi_filter *filter, const char *name)
```

Comment:
Okay, I've tried to read this over and over, and I've come to the conclusion that I think we need comments above the new calls that describe how the new functions' filters work.

Reply:
I have added comments to the two new "apply_xxx_filter" functions. Hope that helps.